src/api/controller/api/facet.js

```diff
@@ -2,28 +2,83 @@ import Koa from 'koa';
-    const data = await ctx.publishedFacet.findValuesForField({
-        field: name,
-        filter: typeof filter === 'function' ? undefined : filter,
-        page,
-        perPage,
-        sortBy,
-        sortDir,
-    });
-    const total = await ctx.publishedFacet.countValuesForField(
-        name,
-        typeof filter === 'function' ? undefined : filter,
-    );
-    ctx.body = {
-        data: data.map((d) => ({ value: d.value, count: d.count, id: d._id })),
-        total,
-    };
+    try {
+        const { page = 0, perPage = 10, sortBy, sortDir } = ctx.request.query;
+
+        // Validate parameters
+        const pageNumRaw = parseInt(page, 10);
+        const pageNum =
+            Number.isFinite(pageNumRaw) && pageNumRaw >= 0 ? pageNumRaw : 0;
+        const perPageNumRaw = parseInt(perPage, 10);
+        const perPageNum =
+            Number.isFinite(perPageNumRaw) && perPageNumRaw >= 1
+                ? Math.min(100, perPageNumRaw)
+                : 10;
+
+        // Decode filter if it exists
+        let searchFilter = filter;
+        if (typeof filter === 'string' && filter.length > 0) {
+            try {
+                searchFilter = decodeURIComponent(filter).trim();
+            } catch (e) {
+                searchFilter = filter.trim();
+            }
+        } else if (filter === null || filter === undefined) {
+            searchFilter = undefined;
+        }
+
+        const [data, total] = await Promise.all([
+            ctx.publishedFacet.findValuesForField({
+                field: name,
+                filter: searchFilter,
+                page: pageNum,
+                perPage: perPageNum,
+                sortBy,
+                sortDir,
+            }),
+            ctx.publishedFacet.countValuesForField(name, searchFilter),
+        ]);
+
+        ctx.body = {
+            data: data.map((d) => ({
+                value: d.value,
+                count: d.count,
+                id: d._id,
+            })),
+            total,
+        };
+    } catch (error) {
+        ctx.status = 500;
+        ctx.body = {
+            error: 'Internal server error',
+            message:
+                process.env.NODE_ENV === 'development'
+                    ? error.message
+                    : undefined,
+        };
+    }
 };
 
 const app = new Koa();
 
-app.use(route.get('/:name/:filter', getFacetFilteredValues));
+// Route for the filters
+app.use(
+    route.get('/:name/(.*)', (ctx, name, filter) => {
+        let cleanFilter = filter;
+        if (typeof cleanFilter === 'string') {
+            // Remove leading and trailing slashes manually
+            while (cleanFilter.startsWith('/')) {
+                cleanFilter = cleanFilter.slice(1);
+            }
+            while (cleanFilter.endsWith('/')) {
+                cleanFilter = cleanFilter.slice(0, -1);
+            }
+        }
+        return getFacetFilteredValues(ctx, name, cleanFilter);
+    }),
```

Comment:
This still can be simplified with:

```suggestion
    route.get('/:name/(.*)', (ctx, name, filter = '') => {
        const cleanFilter = filter
            .replace(/^(\/*)/, '') // remove all leading slashes
            .replace(/(\/*)$/, ''); // remove all slashes at the end

        return getFacetFilteredValues(ctx, name, cleanFilter);
    }),
```

Reply:
This create a codeQL error "Polynomial regular expression used on uncontrolled data"